### PR TITLE
Introduce Core module

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
@jpkrohling could you please review? Plz. do not merge yet, I assume it will fail on Windows/Appveyor

**Comment by ppalaga** [comment]
901e1a0 should work on windows too.

A bit of background for this change:

I am hereby introducing the `srcdeps-core` lib that can be used independently by several "frontends",  such as Maven or Gradle. That is the long term goal.

The short term motivation was to fully decouple the build logic from the Maven plugin API, so that I can try other strategies how `srcdeps-core` could be invoked from within Maven. ATM `srcdeps` acts as a Maven extension that implements `org.apache.maven.AbstractMavenLifecycleParticipant`. As we (me and @jpkrohling ) know, this works for most of dependencies with the notable exception of parent projects and dependencyManagement imports. To be able to support `-SRC-` versions also in parent projects and dependencyManagement imports, I want to try to somehow hook into the local maven repository API. Such experiments will be much easier with a decoupled `srcdeps-core` lib.

The trade off of the present change is that I dropped the support for all SCMs except for Git. But it should be easy to add them back later.

**Comment by ppalaga** [comment]
@jamezp could you please review? @jpkrohling has no time ATM

File: srcdeps-core/src/main/java/org/l2x6/srcdeps/core/BuildRequest.java [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+            this.verbosity = verbosity;
+            return this;
+        }
+    };
+
+    /**
+     * The verbosity level to use when configuring the {@link Builder}.
+     */
+    public enum Verbosity {
+        debug, default_("default"), quiet;
```
Comment (jamezp) [inline srcdeps-core/src/main/java/org/l2x6/srcdeps/core/BuildRequest.java]: Just an idea, but would `info` or `normal` work so you don't need the special character at the end?

File: srcdeps-core/src/main/java/org/l2x6/srcdeps/core/BuildRequest.java [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        public static Verbosity ofId(String id) {
+            switch (id) {
```
Comment (jamezp) [inline srcdeps-core/src/main/java/org/l2x6/srcdeps/core/BuildRequest.java]: Maybe convert the case to make it case insesnitive.

``` java
switch (id.toLowerCase(Locale.ROOT));
```

**Comment by jamezp** [comment]
Looks good to me @ppalaga .

File: srcdeps-core/src/main/java/org/l2x6/srcdeps/core/BuildRequest.java (hunk 1 above)
Comment (ppalaga) [inline srcdeps-core/src/main/java/org/l2x6/srcdeps/core/BuildRequest.java]: Sounds as a nice trick, thanks for the idea :)

I must say, I took the three levels quite mechanically from what maven supports on command line ( `--debug`, nothing, `--quiet`). Thinking of it again, (at least for maven 3.4) it should actually be possible to configure the slf4j logger used by maven using the standard trace, debug, info, warn and error levels. In any case, I should add some JavaDoc to the individual levels.

File: srcdeps-core/src/main/java/org/l2x6/srcdeps/core/BuildRequest.java (hunk 2 above)
Comment (ppalaga) [inline srcdeps-core/src/main/java/org/l2x6/srcdeps/core/BuildRequest.java]: +1

**Comment by ppalaga** [comment]
@jamezp thanks for the review!

**Pull request merged**